jina/types/arrays/search_ops.py

```diff
@@ -73,3 +74,21 @@ def find(
                 filtered.append(self[pos])
 
         return filtered
+
+    def sample(self, k: int, seed: int = None) -> 'DocumentArray':
+        """random sample k elements from :class:`DocumentArray` without replacement.
+
+        :param k: Number of elements to sample from the document array.
+        :param seed: initialize the random number generator, by default is None. If set will
+            save the state of the random function to produce certain outputs.
+        :return: A sampled list of :class:`Document` represented as :class:`DocumentArray`.
+        """
+        from .document import DocumentArray
+
+        if k > len(self):
+            raise ValueError('DocumentArray do not have enough Document to sample')
+        if seed:
```

Comment:
seed =0 wont trigger this line, so has to be:

```suggestion
        if seed is not None:
```

Reply:
fixed